prov/gni/src/gnix_vc.c

```diff
@@ -281,34 +281,34 @@ static void __gnix_vc_pack_conn_req(char *sbuf,
 	      sizeof(gni_mem_handle_t);
 	assert(len <= GNIX_CM_NIC_MAX_MSG_SIZE);
 
-	memcpy(cptr, &rtype, sizeof(rtype));
+	*((uint8_t *)cptr) = rtype;
 	cptr += sizeof(rtype);
```

Comment:
Agreed.  I will add a marco for this.
> On May 9, 2016, at 9:46 AM, James Swaro <notifications@github.com> wrote:
> 
> In prov/gni/src/gnix_vc.c <https://github.com/ofi-cray/libfabric-cray/pull/812#discussion_r62510382>:
> 
> > @@ -281,34 +281,34 @@ static void __gnix_vc_pack_conn_req(char *sbuf,
> >  	      sizeof(gni_mem_handle_t);
> >  	assert(len <= GNIX_CM_NIC_MAX_MSG_SIZE);
> >  
> > -	memcpy(cptr, &rtype, sizeof(rtype));
> > +	*((uint8_t *)cptr) = rtype;
> >  	cptr += sizeof(rtype);
> This might not be relevant to this commit, but why add the possibility of typo errors? We should make this into
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub <https://github.com/ofi-cray/libfabric-cray/pull/812/files/abc41c6cb76334b1f6e1fdc1a24af7a70c343be6#r62510382>